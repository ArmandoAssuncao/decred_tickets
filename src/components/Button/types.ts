export interface IProps {
    m: string
    onPress: () => void
    title: string
}

export interface IState {
}
