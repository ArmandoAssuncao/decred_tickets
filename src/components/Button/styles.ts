import { ViewStyle } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'

export default EStyleSheet.create({
    container: {
        backgroundColor: '#2971FF', // get color from theme
        justifyContent: 'center',
        marginLeft: 10,
        paddingHorizontal: 15,
    } as ViewStyle,
})
