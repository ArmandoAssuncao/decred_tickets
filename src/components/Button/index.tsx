import React from 'react'
import { TouchableOpacity } from 'react-native'

import { IProps, IState } from './types'
import styles from './styles'

import { Text } from 'components'

export default class Card extends React.Component<IProps, IState> {
    public static defaultProps: Partial<IProps> = {
        onPress: null,
    }

    render(): JSX.Element {
        const { m, onPress, title } = this.props

        // TODO: get color from theme
        return (
            <TouchableOpacity
                onPress={onPress}
                style={styles.container}
            >
                <Text m={m} style={{color: 'white'}}>{title}</Text>
            </TouchableOpacity>
        )
    }
}
