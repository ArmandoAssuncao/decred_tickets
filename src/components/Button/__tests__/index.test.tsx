import React from 'react'
import 'react-native'
import { shallow } from 'enzyme'

import Button from '../index'

describe('Button', () => {
    const wrapper = shallow(<Button title={'title'} m={'title default'} onPress={jest.fn()} />)

    it('match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('has Text component', () => {
        expect(wrapper.find('Text')).toHaveLength(1)
    })

    it('has TouchableOpacity component', () => {
        expect(wrapper.find('TouchableOpacity')).toHaveLength(1)
    })

    describe('actions', () => {
        it('call onPress when pressed', () => {
            wrapper.props().onPress()
            expect(wrapper.prop('onPress')).toBeCalled()
        })
    })
})
