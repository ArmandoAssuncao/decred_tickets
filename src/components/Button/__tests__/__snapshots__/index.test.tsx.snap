// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button match snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#2971FF",
      "justifyContent": "center",
      "marginLeft": 10,
      "paddingHorizontal": 15,
    }
  }
>
  <Text
    m="title default"
    size="md"
    style={
      Object {
        "color": "white",
      }
    }
    type="normal"
  >
    title
  </Text>
</TouchableOpacity>
`;
